### PR TITLE
Gen AI: Add "learn more" link to error message when unauthorized

### PR DESCRIPTION
**Pull request by bencodeorg**
Adds a link to a Zendesk article with Gen AI FAQ when you try to submit a chat when unauthorized. Also makes a slight modification to our design system Alert component to allow styling to force the link to be on a single line (see context in [this thread](https://codedotorg.slack.com/archives/C06FELVTV0Q/p1729190737181309?thread_ts=1729021564.344969&cid=C06FELVTV0Q)).

**After this change (no "Learn more" link previously)**

image

## Links

- jira ticket: [LABS-1118](https://codedotorg.atlassian.net/browse/LABS-1118)

## Testing story

Tested as a signed out user. Also tested that a error alert like the one we display if your model customizations failed to save does not display this link (I had to comment out the toxicity detection to test this out).

## Follow-up work

Planning on copying over these same error messages and display them if you try to submit a model customization without being authorized.

### CODE REVIEW

**Review by sanchitmalhotra126 — APPROVED** [review]
Nice!

File: apps/src/componentLibrary/link/Link.tsx [hunk 1]
```diff
@@ -31,12 +31,14 @@ export interface LinkBaseProps extends HTMLAttributes<HTMLAnchorElement> {
 export type LinkWithChildren = LinkBaseProps & {
   /** Link content */
   children: React.ReactNode;
+  className?: string;
```
Comment (levadadenys) [inline apps/src/componentLibrary/link/Link.tsx:34]: @bencodeorg LinkBaseProps already have optional classname prop. Why do we need this here?
Also, if adding anything to DSCO components - you need to update CHANGELOG.md for that component and it's preferable to make DSCO components changes a separate PR since that'll be much easier for keeping track of changes and reverts if needed

File: apps/src/componentLibrary/link/Link.tsx [hunk 2]
```diff
@@ -31,12 +31,14 @@ export interface LinkBaseProps extends HTMLAttributes<HTMLAnchorElement> {
...
   text?: never;
 };
 
 export type LinkWithText = LinkBaseProps & {
   /** Link text content */
   text: string;
+  className?: string;
```
Comment (levadadenys) [inline apps/src/componentLibrary/link/Link.tsx:41]: @bencodeorg LinkBaseProps already have optional classname prop. Why do we need this here?

File: apps/src/componentLibrary/link/Link.tsx (hunk 1 above)
Comment (bencodeorg) [reply apps/src/componentLibrary/link/Link.tsx:34]: 🤦 My bad, misread the type definition. Fixed here!

https://github.com/code-dot-org/code-dot-org/pull/61915
